### PR TITLE
build: making it so ephemeral list-type arguments are appended to existing list-type arguments

### PR DESCRIPTION
**Pull request by jbohren**
This allows you to add additional CMake arguments on top of the arguments in a given profile.

@adeguet1 does this do what you needed?

### CODE REVIEW

File: catkin_tools/context.py [hunk 1]
```diff
@@ -126,9 +128,12 @@ def Load(cls, workspace_hint=None, profile=None, opts=None, strict=False):
 
         # User-supplied args are used to update stored args
         # Only update context args with given opts which are not none
-        context_args.update(dict([
-            (k, v) for (k, v) in opts_vars.items()
-            if k in Context.KEYS and v is not None]))
+        for (k,v) in opts_vars.items():
+            if k in Context.KEYS and v is not None:
+                if append and type(context_args.get(k,None)) is list and type(v) is list:
```
Comment (wjwwood) [inline catkin_tools/context.py]: nitpick: `k, None`

File: catkin_tools/context.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ class Context(object):
         'space_suffix']
 
     @classmethod
-    def Load(cls, workspace_hint=None, profile=None, opts=None, strict=False):
+    def Load(cls, workspace_hint=None, profile=None, opts=None, strict=False, append=False):
```
Comment (wjwwood) [inline catkin_tools/context.py]: nitpick: I guess I missed this on my original review, but methods should always by lowercase: https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables

Comment (jbohren) [inline catkin_tools/context.py]: Even class methods?

File: catkin_tools/context.py (hunk 1 above)
Comment (jbohren) [inline catkin_tools/context.py]: Also `k,v` above.

File: catkin_tools/context.py (hunk 2 above)
Comment (jack-oquin) [inline catkin_tools/context.py]: Yes, even class methods. :smile:

Comment (jbohren) [inline catkin_tools/context.py]: I'll change that, then.

**Comment by jbohren** [comment]
@wjwwood I'll update the `Load` / `Save` functions in another PR.

**Pull request merged**